File: PP/Java/P1Stiva/src/stiva/Stiva.java
package stiva;

public class Stiva implements InterfaceStiva{
	private Nod Head;
	Stiva(){
		Head=null;
	}
	public void push(int val) {
		Nod p=new Nod(val);
		p.leg=Head;
		Head=p;
		
	}

	public int top() {
		if(Head==null)
		{
			System.err.println("Stiva este goala");
			return 0;
		}
		else
		{
			return Head.data;
		}
	}

	public int pop() {
		if(Head==null)
		{
			System.err.println("Stiva este goala");
			return 0;
		}
		else
		{
			int aux=Head.data;
			//Nod p=Head;
			Head=Head.leg;
			return aux;
		}
	}

	public boolean isEmpty() {
		if(Head==null)
			return true;
		else
			return false;
	}
	
}

File: SD/Toate functiile+teste+subiecte examen/Arbori Binari/Arbori Binari/stiva.cpp
#include"stiva.h"
void initS(stiva *&S)
{
	S = nullptr;
}
void push(stiva *&S, Nod* val)
{
	stiva *p = new stiva;
	if (p == 0)
		return;
	p->data = val;
	p->leg = S;
	S = p;
}
Nod *pop(stiva *&S)
{
	if (!S)
	{
		cout << "\nStiva este goala";
		return 0;
	}
	Nod *aux = S->data;
	stiva *p = S;
	S = S->leg;
	delete p;
	return aux;
}